dis_snek/annotations/argument_annotations.py

```diff
@@ -0,0 +1,68 @@
+from typing import Union, TYPE_CHECKING, Any, List, Type, Dict
+
+from dis_snek.const import MISSING
+from dis_snek.utils.misc_utils import get_parameters
+
+from dis_snek.models.context import Context
+from dis_snek.models.scale import Scale
+from dis_snek.client import Snake
+
+if TYPE_CHECKING:
+    from dis_snek.models import Member, User, TYPE_MESSAGEABLE_CHANNEL
+
+
+def define_annotation():
+    """
+    Define a function as an annotation.
+
+    The function **must** type hint its expected arguments. Make sure your annotation runs quickly,
+    as command execution will be delayed until all annotations are completed
+
+    **Supported Types:**
+    `Context`
+    `Scale`
+    """
+
+    def wrapper(func):
+        params = get_parameters(func)
+        args = []
+        for param in params.values():
+            if param.annotation is Context:
```

Comment:
I think it would be better to check with `issubclass()`, so users can use `InteractionContext` etc for their type hinting

Reply:
Will be resolved in a follow-up commit